### PR TITLE
Wip refactoring

### PR DESCRIPTION
**Pull request by JoaoPi314**

### CODE REVIEW

File: src/com/melqjpgames/entities/Entity.java [hunk 1]
```diff
@@ -1,26 +1,68 @@
...
+	 */
+	protected boolean damaged;
+	/**
+	 * Number of frames the damage sprite should be rendered 
+	 */
+	protected int damageFrames;
+	/**
+	 * Knockback speed when taking damage
+	 */
+	protected int kbSpeed;
```
Comment (kmmelcher) [inline src/com/melqjpgames/entities/Entity.java]: knockbackSpeed

**Review by kmmelcher — COMMENTED** [review]
You are doing great!

File: src/com/wellmax/entities/Skeleton.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import com.wellmax.entities.types.Directions;
+import com.wellmax.main.Game;
+
+/**
+ * Skeleton class. The skeleton fires bones when player is near. 
+ * It also is stronger than orcs
+ * @author joao.gomes
+ *
+ */
+public class Skeleton extends Enemy{
```
Comment (kmmelcher) [inline src/com/wellmax/entities/Skeleton.java:14]: I liked this!

File: src/com/wellmax/entities/types/Directions.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.wellmax.entities.types;
+
+public enum Directions {
```
Comment (kmmelcher) [inline src/com/wellmax/entities/types/Directions.java:3]: This is also a really good architecture

File: todo.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 -- refactoring (MUST DO FIRST)
-	-- Implement collidsion check with entity to stop using wallTile to everythiing with collision
+	-- Adjust collision mask of trees
+	-- Implement collision check with entity to stop using wallTile to everythiing with collision
```
[the remark this answers is not in the document]
Comment (JoaoPi314) [reply todo.txt:3]: I forgot to remove this file :smile:. When I finish refactoring I will create some issues with new mechanics to the Game.

File: src/com/melqjpgames/entities/GenericEntity.java [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+	 */
+	protected int maskX;
+	/**
+	 * y position of collision mask(offset based on y attribute)
+	 */
+	protected int maskY;
+	/**
+	 * Width of collision mask. Based on (maskX, maskY) point
+	 */
+	protected int mWidth;
```
[the remark this answers is not in the document]
Comment (JoaoPi314) [reply src/com/melqjpgames/entities/GenericEntity.java]: Changed

File: src/com/melqjpgames/entities/GenericEntity.java [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+		this.mWidth = mWidth;
+	}
+	/**
+	 * Setter method to mHeight attribute
+	 * @param mHeight mask height
+	 */
+	public void setMHeight(int mHeight) {
+		this.mHeight = mHeight;
+	}
+
```
[the remark this answers is not in the document]
Comment (JoaoPi314) [reply src/com/melqjpgames/entities/GenericEntity.java]: I removed all the getter and setter comments

**Pull request merged**